### PR TITLE
removed default config function

### PR DESCRIPTION
**Pull request by xh3b4sd**
Hashtag boyscouting.

### CODE REVIEW

File: framework/default.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package framework
```
Comment (xh3b4sd) [inline framework/default.go]: I put all the default functions in one file. I like this separation.

File: framework/framework.go [hunk 2]
```diff
@@ -24,8 +24,6 @@ import (
 
 // Config represents the configuration used to create a new operator framework.
 type Config struct {
-	// Dependencies.
-
```
Comment (xh3b4sd) [inline framework/framework.go]: I removed the comments stating the things obvious for us.

File: framework/framework.go [hunk 3]
```diff
@@ -43,49 +41,24 @@ type Config struct {
...
-		ResourceRouter: nil,
-
-		// Settings.
-		BackOffFactory: func() backoff.BackOff {
-			b := backoff.NewExponentialBackOff()
-			b.MaxElapsedTime = 0
-			return backoff.WithMaxTries(b, 7)
-		},
-	}
-}
```
Comment (xh3b4sd) [inline framework/framework.go]: This got removed since we agreed on not having these anymore.

File: framework/framework.go [hunk 4]
```diff
@@ -99,25 +72,21 @@ func New(config Config) (*Framework, error) {
 		return nil, microerror.Maskf(invalidConfigError, "config.ResourceRouter must not be empty")
 	}
 
-	// Settings.
 	if config.BackOffFactory == nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.BackOffFactory must not be empty")
+		config.BackOffFactory = DefaultBackOffFactory()
```
Comment (xh3b4sd) [inline framework/framework.go]: Applying defaults.

File: framework/framework.go [hunk 5]
```diff
@@ -99,25 +72,21 @@ func New(config Config) (*Framework, error) {
...
 		resourceRouter: config.ResourceRouter,
 
-		// Settings.
-		backOffFactory: config.BackOffFactory,
-
-		// Internals.
 		bootOnce: sync.Once{},
 		mutex:    sync.Mutex{},
+
+		backOffFactory: config.BackOffFactory,
```
Comment (xh3b4sd) [inline framework/framework.go]: The order is dependencies, internals, settings. Hashtag alphabet.

File: framework/resource_set.go [hunk 6]
```diff
@@ -33,12 +33,6 @@ type ResourceSetConfig struct {
 	Resources []Resource
 }
 
-func DefaultResourceSetResourceFunc(rs []Resource) func(ctx context.Context, obj interface{}) ([]Resource, error) {
-	return func(ctx context.Context, obj interface{}) ([]Resource, error) {
-		return rs, nil
-	}
-}
```
Comment (xh3b4sd) [inline framework/resource_set.go]: This got moved to `default.go`.

File: framework/framework.go (hunk 5 above)
Comment (kopiczko) [reply framework/framework.go]: Alphabet collides with logic here. Internals are constructed from dependencies and settings and I think it's easier to maintain it like that.

In New you validate dependencies and settings first, so it's easier to not miss any if they aren't intertwined with settings

Comment (kopiczko) [reply framework/framework.go]: This is also documented here: https://github.com/giantswarm/fmt/pull/4/files#diff-09196e856cd1c9274e7747be39ebfbdbR200

File: framework/framework.go (hunk 2 above)
Comment (kopiczko) [reply framework/framework.go]: No strong feelings.

File: framework/framework.go (hunk 5 above)
Comment (xh3b4sd) [reply framework/framework.go]: It was always dependencies, internals, settings from the very beginning. It looks like I did not pay that much attention to the formatting rules we documented or I missed that detail. 

> Internals are constructed from dependencies and settings

This does only represent some cases. As you see here, this is not the case.

Comment (kopiczko) [reply framework/framework.go]: Yes, but in general it's good to see what is copied directly from Config struct. That's practical thing.

Comment (xh3b4sd) [reply framework/framework.go]: I think you see what is copied. The order does not matter for this to recognize. Lets talk next week so we can agree on something for the long run.

**Pull request merged**